scripts/NewznabToCardigannYML.ps1

```diff
@@ -35,208 +48,273 @@ function Invoke-NameReplace {
+}
+function Invoke-YMLReplace
+{
+    param (
+        [Parameter(Mandatory)]
+        [string]
+        $Name
+    )
+    return ($Name -replace 'category2', 'category' -replace '"{ ', '{ ' -replace ' }"', ' }')
+}
 # Get Data and digest objects
+Write-Information 'Requesting Caps'
 [xml]$xmlResponse = (Invoke-WebRequest -Uri $capscall -Headers $headers -ContentType 'application/xml' -Method Get).Content
 $rSearchCaps = $xmlResponse.caps.searching
 $rCategories = $xmlResponse.caps.categories.category
-
+$rServer = $xmlResponse.caps.server
+Write-Information 'Got Caps'
 # Caps
-[string]$q_search = "[$($rSearchCaps.search.supportedParams -replace ",",", ")]"
-[string]$movie_search = "[$($rSearchCaps.'movie-search'.supportedParams -replace ",",", ")]"
-[string]$tv_search = "[$($rSearchCaps.'tv-search'.supportedParams -replace ",",", ")]"
-[string]$book_search = "[$($rSearchCaps.'book-search'.supportedParams -replace ",",", ")]"
-[string]$audio_search = "[$($rSearchCaps.'audio-search'.supportedParams -replace ",",", ")]"
-
-# Get Main Categories: ID, Name
+[string]$q_search = "[$($rSearchCaps.search.supportedParams -replace 'group, ',''  -replace ', ',',' )]"
+[string]$movie_search = "[$($rSearchCaps.'movie-search'.supportedParams -replace ', ',',')]"
+[string]$tv_search = "[$($rSearchCaps.'tv-search'.supportedParams -replace ', ',',')]"
+[string]$book_search = "[$($rSearchCaps.'book-search'.supportedParams -replace ', ',',')]"
+[string]$audio_search = "[$($rSearchCaps.'audio-search'.supportedParams -replace ', ',',')]"
+Write-Information 'Search Caps Built'
+# Get Categories: ID, MappedName, Name
 [System.Collections.Generic.List[System.Object]]$categories = @()
-
+Write-Information 'Building Categories'
 foreach ($category in $rCategories)
 {
-    $catName = Invoke-NameReplace -Name $category.name
+    $catName = Invoke-CatNameReplace -Name $category.name
     $temp = [PSCustomObject][ordered]@{
-        id = $category.id
-        cat = $catName
-        desc = $category.description
+        id   = $category.id
+        cat  = $catName
+        desc = "'$($category.name)'"
     }
     $categories.Add($temp)
-}
-
-foreach ($category in $rCategories.subcat)
-{
-    $catName = Invoke-NameReplace -Name "$($category.description)/$($category.name)"
-    $temp = [PSCustomObject][ordered]@{
-        id = $category.id
-        cat = $catName
-        desc = $category.description
+    Write-Information 'Building Sub-Categories'
+    foreach ($subcategory in $category.subcat)
+    {
+        $subcatName = Invoke-CatNameReplace -Name "$($catName)/$($subcategory.name)"
+        $subtemp = [PSCustomObject][ordered]@{
+            id   = $subcategory.id
+            cat  = $subcatName
+            desc = "'$($catName)/$($subcategory.name)'"
+        }
+        $categories.Add($subtemp)
     }
```

Comment:
If merging these two you may as well get rid of the $ymlcategories bit and do it all here